src/cartridge_header.rs

```diff
@@ -1,51 +1,219 @@
+            not_used,
+            joybus_mode_entry_point,
         }
     }
 
-    fn extract_entry_point(data: &[u8]) -> [u8; 4] {
+    // 32bit ARM branch opcode, eg. "B rom_start"
+    fn extract_rom_entry_point(data: &[u8]) -> [u8; 4] {
         data[0x000..=0x003]
             .try_into()
-            .expect("extracting entry point")
+            .expect("extracting rom entry point")
     }
 
+    // compressed bitmap, required!
     fn extract_nintendo_logo(data: &[u8]) -> [u8; 156] {
         data[0x004..=0x09F]
             .try_into()
             .expect("extracting nintendo logo")
     }
 
+    // uppercase ascii, max 12 characters
     fn extract_game_title(data: &[u8]) -> String {
         let game_title_bytes: [u8; 12] = data[0x0A0..=0x0AB]
             .try_into()
             .expect("extracting game title");
-        String::from_utf8(game_title_bytes.into()).expect("parsing game title")
+        
+        String::from_utf8(game_title_bytes.into())
+            .expect("parsing game title")
     }
 
+    // uppercase ascii, 4 characters
     fn extract_game_code(data: &[u8]) -> String {
         let game_code_bytes: [u8; 4] = data[0x0AC..=0x0AF]
             .try_into()
             .expect("extracting game code");
-        String::from_utf8(game_code_bytes.into()).expect("parsing game code")
+        
+        String::from_utf8(game_code_bytes.into())
+            .expect("parsing game code")
+    }
+
+    
+    // uppercase ascii, 2 characters
+    fn extract_marker_code(data: &[u8]) -> String {
+        let marker_code_bytes: [u8; 2] = data[0x0B0..=0x0B1]
+            .try_into()
+            .expect("extracting marker code");
+        
+        String::from_utf8(marker_code_bytes.into())
+            .expect("parsing marker code")
+    }
+
+    // must be 96h, required!
+    fn extract_fixed_value(data: &[u8]) -> [u8; 1] {
+        data[0x0B2..=0x0B2]
+            .try_into()
+            .expect("extracting fixed value")
+    }
+
+    // 00h for current GBA models
+    fn extract_main_unit_code(data: &[u8]) -> [u8; 1] {
+        data[0x0B3..=0x0B3]
+            .try_into()
+            .expect("extracting main unit code")
+    }
+
+    // usually 00h (bit7=DACS/debug related)
+    fn extract_device_type(data: &[u8]) -> [u8; 1] {
+        data[0x0B4..=0x0B4]
+            .try_into()
+            .expect("extracting device type")
+    }
+
+    
+    // should be zero filled
+    fn extract_reserved_area_1(data: &[u8]) -> [u8; 7] {
+        data[0x0B5..=0x0BB]
+            .try_into()
+            .expect("extracting reserved area 1")
+    }
+
+    // usually 00h
+    fn extract_software_version(data: &[u8]) -> [u8; 1] {
+        data[0x0BC..=0x0BC]
+            .try_into()
+            .expect("extracting software version")
+    }
+
+    // header checksum, required!
+    fn extract_complement_check(data: &[u8]) -> [u8; 1] {
+        // TODO: Do we check if extract_complement_check header (data[0x0BD..=0x0BD]) is correct?
```

Comment:
Let's remove this and open a issue, right now is not a problem but is that check exists we should consider it in the future :)

Reply:
I've just done it!